File: xjlib/src/main/java/me/xjliao/xjlib/net/XResponse.java
/*
 * Copyright (c) 2017 xjliao.me created by xjliao
 * ProjectName: xjl
 * ModuleName: xjlib
 * FileName: XResponse.java
 * ClassName: XResponse
 * LastModified: 11/29/17 9:33 AM
 */

package me.xjliao.xjlib.net;

import com.fasterxml.jackson.annotation.JsonProperty;

public class XResponse<T> {

    private int code;

    @JsonProperty(value = "message")
    private String msg;

    private T obj;

    public XResponse() {

    }

    public XResponse(int code, String msg) {
        this.code = code;
        this.msg = msg;
    }

    public XResponse(int code, String msg, T obj) {
        this(code, msg);
        this.obj = obj;
    }

    public int getCode() {
        return code;
    }

    public void setCode(int code) {
        this.code = code;
    }

    public String getMsg() {
        return msg;
    }

    public void setMsg(String msg) {
        this.msg = msg;
    }

    public T getObj() {
        return obj;
    }

    public void setObj(T obj) {
        this.obj = obj;
    }
}

File: xjlib/src/main/java/me/xjliao/xjlib/net/XResponse.kt
/*
 * Copyright (c) 2017 xjliao.me created by xjliao
 * ProjectName: xjl
 * ModuleName: xjlib
 * FileName: XResponse.java
 * ClassName: XResponse
 * LastModified: 12/14/17 11:26 AM
 */

package me.xjliao.xjlib.net

import com.fasterxml.jackson.annotation.JsonProperty

class XResponse<T> {

    var code: Int = 0

    @JsonProperty(value = "message")
    var msg: String? = null

    var obj: T? = null

    constructor() {

    }

    constructor(code: Int, msg: String) {
        this.code = code
        this.msg = msg
    }

    constructor(code: Int, msg: String, obj: T) : this(code, msg) {
        this.obj = obj
    }

    companion object {
        val DEFAULT = "DEFAULT"
    }
}